lib/File/stat.pm

```diff
@@ -199,15 +199,16 @@ sub lstat :prototype($) { populate(CORE::lstat(shift)) }
 sub stat :prototype($) {
     my $arg = shift;
     my $st = populate(CORE::stat $arg);
-    return $st if defined $st;
-	my $fh;
+    return $st if defined $st || ref $arg;
```

Comment:
Maybe out of scope of this PR, but this function is context sensitive on the other failure returns: the later return statement as well as the return value of the final populate(). Should probably force scalar return values (as well as in lstat)

Reply:
Good point, but for this PR I went the other way: Always return an empty list for failure in order to be 100% compatible with any existing callers that may use `stat` in list context (even though this case is undocumented and untested, apparently).